### ISSUE TITLE
Add Kibana Reporting diagnostics

### ISSUE THREAD

**Issue by tsullivan** [elastic/support-diagnostics#676]
Reporting has a self-service diagnostic tool in Stack Management > Reporting. 

From [the docs](https://www.elastic.co/guide/en/kibana/current/reporting-troubleshooting.html#:~:text=Reporting%20diagnosticsedit&text=When%20Kibana%20is%20running%2C%20navigate,up%20with%20any%20relevant%20recommendations.):
> [In the] the Report Listing page, and click Run reporting diagnostics. This will open up a diagnostic tool that checks various parts of the Kibana deployment and come up with any relevant recommendations

We should have this Reporting Diagnostic get pulled as part of the general Kibana diagnostic. It would help speed up discovery and timelines for the Support team.

**Comment by tsullivan** [comment]
Replaces: https://github.com/elastic/kibana/issues/151033

**Comment by tsullivan** [comment]
The diagnostic tool is under **Stack Management > Alerts and Insights > Reporting > Run reporting diagnostics.**

<img width="1701" alt="image" src="https://github.com/elastic/support-diagnostics/assets/908371/4d6e0853-dd97-4575-8175-48bb9ae3ffe2">

The client request to the Reporting Diagnostic API ("Check browser" functionality):
```
curl 'https://localhost:15601/api/reporting/diagnose/browser' \
  -X 'POST' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en-US,en;q=0.9' \
  -H 'Connection: keep-alive' \
  -H 'Content-Length: 0' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: sid=Fe26.2**9334e4394728b923c095ebc2c2f455100eba49b27e0d9865ea537cbbcea0e430*cb2KojtZqtowPCqMcqOM9g*8rb5CArN-U1RK85kKWXPfdWRK2rA4ZKN3sweGjUpvTqvWoH2KTmp-FzPzL0nzOztFZqzhAACdQNxZWI3dL6FvHTEm7RNlekCoS1wOaw1WB3tKS8cTM0cQaKoa4kro3wzTUAP0JvYEto_xJ5uXa-NdAKMStRe9n0tH7ih6lVrPMQiKe4RmwMLGbAzJt09akFRxymqyAwQZ28aKw5lZ6Zds3WnhJ46uyPyAVWlbOA5xucnJoYzw2jODJGOatI_xtBe**24917e388462cd45644a68c93dac0382daefd5d044d65354f3726dce347a1f92*NaupgX2UNWsSb0w3Oi-2UVRjVqhQyiodTv1STYWUXys' \
  -H 'Origin: https://localhost:15601' \
  -H 'Referer: https://localhost:15601/app/management/insightsAndAlerting/reporting' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36' \
  -H 'kbn-system-request: true' \
  -H 'kbn-version: 8.8.2' \
  -H 'sec-ch-ua: "Google Chrome";v="119", "Chromium";v="119", "Not?A_Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'x-kbn-context: %7B%22type%22%3A%22application%22%2C%22name%22%3A%22management%22%2C%22url%22%3A%22%2Fapp%2Fmanagement%2FinsightsAndAlerting%2Freporting%22%7D' \
  --compressed \
  --insecure
```

**Comment by stefnestor** [comment]
@tsullivan quick mentioning I believe historically this repo attempted to avoid POST requests ( right, @pickypg ?); is there a way to poll this endpoint as GET?

**Comment by pickypg** [comment]
Correct, we don't support any `POST` endpoints.

**Comment by tsullivan** [comment]
I believe this endpoint doesn't need to be a POST. I have filed https://github.com/elastic/kibana/issues/171698 to have the endpoint changed to GET.

**Comment by tsullivan** [comment]
@pickypg the endpoint has been [changed to GET](https://github.com/elastic/kibana/pull/192232), for 8.16.0+.

What would our next steps be on this issue?

cc @elastic/appex-sharedux

**Comment by pickypg** [comment]
It seems like the only thing necessary to make the request work is to call it with the `kbn-version` specified?

**Comment by tsullivan** [comment]
> It seems like the only thing necessary to make the request work is to call it with the kbn-version specified?

That might not even be necessary, for our stateful deployments. I was able to get the response from an 8.16.0-SNAPSHOT deployment running in Docker using this command:

```
curl 'https://localhost:15601/internal/reporting/diagnose/browser' \
  -k -u elastic:changeme
```

response:
```
{"success":true,"help":[],"logs":"\nDevTools listening on ws://127.0.0.1:33695/devtools/browser/bc1bd0a3-fed1-4f90-8d99-e46d0fdbae98\n[0921/184802.861275:WARNING:sandbox_linux.cc(430)] InitializeSandbox() called with multiple threads in process gpu-process.\n"}%
```
